Scripts/Enemy_Scripts/enemy.gd

```diff
@@ -0,0 +1,128 @@
+extends CharacterBody3D
+
+# Three states, each of which are controlled by their own respective functions.
+enum EnemyState { PATROL, CHASE, SEARCH }
+
+@export var speed: float = 3.0
+@export var patrol_speed: float = 1.5
+@export var chase_speed: float = 3.0
+@export var detection_radius: float = 10.0
+@export var debug_mode: bool = false
+
+@export var patrol_min_bounds: Vector3 = Vector3(-50, 0, -15)
+@export var patrol_max_bounds: Vector3 = Vector3(10, 0, 20)
+
+var player: CharacterBody3D = null
+var state: EnemyState = EnemyState.PATROL
+var b_can_see_player: bool = false
+
+@onready var nav_agent: NavigationAgent3D = $NavigationAgent3D
+@onready var nav_region: NavigationRegion3D = get_node("../NavigationRegion3D")
+@onready var raycast: RayCast3D = $VisionRaycast
+
+func _ready():
```

Comment:
please add return types to functions

func _ready() -> void:

There's an editor setting that will do a lot of these automatically.
Helps with compiler optimisation and safety checks